test_util/launch.py

```diff
@@ -1,20 +1,169 @@
+  delete    Destroying the provided cluster deployment
+
+Options:
+  --config-path=<path>  Path for config to create cluster from [default: config.yaml].
+  --info-path=<path>    JSON file output by create and consumed by wait, describe,
+                        and delete [default: cluster_info.json].
 """
+import abc
+import copy
+import os
+import sys
 
+import yaml
 from docopt import docopt
 
+from pkgpanda.util import json_prettyprint, load_json, load_string, write_json
+
+from test_util.aws import BotoWrapper, DcosCfSimple
+
+
+class LauncherError(Exception):
+    def __init__(self, error, msg):
+        self.error = error
+        self.msg = msg
+
+    def __repr__(self):
+        return '{}: {}'.format(self.error, self.msg)
+
+
+class AbstractLauncher(metaclass=abc.ABCMeta):
+    def create(self, config):
+        raise NotImplementedError()
+
+    def wait(self, info):
+        raise NotImplementedError()
+
+    def describe(self, info):
+        raise NotImplementedError()
+
+    def delete(self, info):
+        raise NotImplementedError()
+
+
+class AwsCloudformationLauncher(AbstractLauncher):
+    def __init__(self, boto_wrapper):
+        self.boto_wrapper = boto_wrapper
+
+    def create(self, config):
+        self.boto_wrapper.create_stack(
+            config['stack_name'], config['template_url'], config['parameters'])
+        return {
+            'type': 'cloudformation',
+            'stack_name': config['stack_name'],
+            'provider_info': {
+                'region': config['provider_info']['region'],
+                'access_key_id': config['provider_info']['access_key_id'],
+                'secret_access_key': config['provider_info']['secret_access_key']}}
+
+    def wait(self, info):
+        # TODO: should this support the case where the cluster is being updated?
+        cf = self.get_stack(info)
+        status = cf.get_stack_details()['StackStatus']
+        if status == 'CREATE_IN_PROGRESS':
+            cf.wait_for_stack_creation(wait_for_poll_min=0)
+        elif status == 'CREATE_COMPLETE':
+            return
+        else:
+            raise LauncherError('WaitError', 'AWS Stack has entered unexpected state: {}'.format(status))
+
+    def describe(self, info):
+        # include the basic info
+        desc = copy.copy(info)
+        cf = self.get_stack(info)
+        desc.update({
+            'masters': convert_host_list(cf.get_master_ips()),
+            'private_agents': convert_host_list(cf.get_private_agent_ips()),
+            'public_agents': convert_host_list(cf.get_public_agent_ips())})
+        return desc
+
+    def delete(self, info):
+        self.get_stack(info).delete()
+
+    def get_stack(self, info):
+        """Returns the correct class interface depending how the AWS CF is configured
+        NOTE: only supports Simple Cloudformation currently
+        """
+        return DcosCfSimple(info['stack_name'], self.boto_wrapper)
+
+
+def get_launcher(launcher_type, provider_info):
```

Comment:
`laucher_type: str, provider_info: dict`   and you could remove those descriptions in the docstrings.